### PR TITLE
add ocaml.4.10.0+nnpcheck to dynamically check for naked pointers

### PR DESCRIPTION
**Pull request by avsm**
see https://github.com/ocaml/ocaml/pull/9534 for more discussion

### CODE REVIEW

**Comment by kayceesrk** [comment]
Great. Thanks!

File: packages/ocaml-variants/ocaml-variants.4.10.0+nnpcheck/opam [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "--prefix=%{prefix}%"
+    "CC=cc"
+    "ASPP=cc -c"
+  ] {os = "openbsd" | os = "freebsd" | os = "macos"}
+  [make "-j%{jobs}%" {os != "cygwin"} "world"]
+  [make "-j%{jobs}%" {os != "cygwin"} "world.opt"]
+]
+install: [make "install"]
+url {
+  src: "git+https://github.com/kayceesrk/ocaml.git#4.10.0+nnp+check"
```
Comment (kit-ty-kate) [inline packages/ocaml-variants/ocaml-variants.4.10.0+nnpcheck/opam:31]:
```suggestion
  src: "https://github.com/kayceesrk/ocaml/archive/4.10.0+nnp+check.tar.gz"
```
We can avoid git here

**Comment by kayceesrk** [comment]
There will be a separate branch for `4.10.1` if there will be one.

**Comment by kit-ty-kate** [comment]
Sweet! Then once the git requirement is removed this should be good to go.

**Comment by avsm** [comment]
The reason I made this git is that it's much more efficient as a pinned repository.  This particular variant is intended for a very specific development use, so I think that's fine. (e.g. if @kayceesrk pushes more changes to this branch as the OCaml PR develops).  There's not too much utility in making this a http archive without a corresponding checksum, which won't work in this case.

(I realise that this is a departure from the normal opam-repository policy, but we've done this in the past for other cases where it's useful to the community, such as with early ppx_tools).

**Comment by kit-ty-kate** [comment]
Ok then. Thanks!

**Comment by kayceesrk** [comment]
Thanks @kit-ty-kate!

**Comment by avsm** [comment]
Thanks @kit-ty-kate!

**Pull request merged**